### PR TITLE
Lower retrieval rate of disabled module output

### PR DESCRIPTION
**Pull request by igorwulff**
This change ensures we retrieve the modules_disable_output data only once and use that for every block output where we need to check if the output is disabled. This improves performance by a bit as well. Only pages that have all cache enabled (excl. FPC) the gains are 5-10m depending on the page, this amount is bigger on pages that is missing block_cache for certain blocks, 30ms

Also the get for getHtml has been changed to getData('html'). There is a minimal improvement to performance, without block cache (it was hit 1700 times) on a PLP the gain is about 6-7ms, and with full block_cache its only 1ms, but it seems like an easy gain.

### Related Pull Requests
https://github.com/magento-gl/magento2ee/pull/158

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by igorwulff** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please review and fix failing unit & static tests

**Comment by igorwulff** [comment]
@ihor-sviziev The static failing tests don't seem to be relevant or perhaps I'm misreading them?

For the Unit Tests.
 I'm thinking about adding a seperate method for the disabled module output check to also make it easier to create mocked return data for the static check in the Unit Test, like this:

```
private function isModuleOutputDisabled(): bool
{
    if (static::$disabledModules === null) {
        static::$disabledModules = $this->_scopeConfig->getValue(
            'advanced/modules_disable_output',
            \Magento\Store\Model\ScopeInterface::SCOPE_STORE
        ) ?? [];
    }

    return isset(static::$disabledModules[$this->getModuleName()])
        && static::$disabledModules[$this->getModuleName()];
}
```

And do something like this in the various Unit Tests:
```
$model = $this->getMockBuilder(Model::class)
    ->setMethods(['isModuleOutputDisabled'])
    ->getMock();

$model->expects($this->once())
    ->method('isModuleOutputDisabled')
    ->willReturn(true);
```

Would this be approach be OK?

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
@magento run unit tests

**Comment by engcom-Delta** [comment]
@magento run unit tests

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@mgento run Unit Tests

**Comment by engcom-Echo** [comment]
@magento run Unit Tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all test

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
Functional Tests B2B test failure are not related to PR.

**Comment by engcom-Echo** [comment]
@ihor-sviziev can you please review PR.

File: lib/internal/Magento/Framework/View/Element/AbstractBlock.php [hunk 1]
```diff
@@ -650,10 +657,17 @@ protected function _beforeToHtml()
...
-            'advanced/modules_disable_output/' . $this->getModuleName(),
-            \Magento\Store\Model\ScopeInterface::SCOPE_STORE
-        )) {
+
+        if (static::$disabledModules === null || !isset(static::$disabledModules[$this->getModuleName()])) {
+            static::$disabledModules[$this->getModuleName()] = $this->_scopeConfig->getValue(
+                'advanced/modules_disable_output/' . $this->getModuleName(),
+                \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+            ) ?? [];
+        }
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/View/Element/AbstractBlock.php:666]: @engcom-Echo Why can't we retrieve all the disabled modules at once, as it was suggested originally?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/View/Element/AbstractBlock.php (hunk 1 above)
Comment (igorwulff) [reply lib/internal/Magento/Framework/View/Element/AbstractBlock.php:666]: That was the original thought out method indeed.

**Comment by engcom-Dash** [comment]
Hello @igorwulff,

Please resolve the conflicts, so that we can move further with this PR for review.

Thanks

**Comment by engcom-Dash** [comment]
@igorwulff - Closing this PR due to no response to the last https://github.com/magento/magento2/pull/36320#issuecomment-2352801481. Feel free to reopen it.

**Pull request closed without merging**